### PR TITLE
Warn when there is less than 1 pixel per contact

### PR DESCRIPTION
**Pull request by sroet**
This PR allows for `**kwargs` to be passed down from `ContactCount.plot()` to `plt.subplots()` giving the user more power over the plotting

~~Also, this makes sure that at least 1 pixel is drawn per contact, even if it overlaps with other contacts Potentially solving #48~~

~~I am not to happy about the current effect that if the initial plot is made with a to low dpi, all the figures saved from that plot also look bad. This might not be the implementation we want~~

This raises a `RuntimeWarning` if the figure is to small to show all the contacts

- [API BREAK] This also combines the `vmin` and `vmax` keyword arguments of the `ContactCount.plot()` function into `vrange=(-1.0, 1.0)` in order to obey codeclimate complexity

### CODE REVIEW

**Comment by sroet** [comment]
This will implement the first option of: https://github.com/dwhswenson/contact_map/issues/48#issuecomment-478257882.

**Comment by sroet** [comment]
@dwhswenson This is ready for review, 
I updated the notebook to use the correct `vrange` instead of `vmin/vmax`
Also added a cell with possible fixes to the warning:
- let jupyter output a pdf
- save as a pdf
- increase figsize
- increase dpi

**Comment by sroet** [comment]
Apparently the `kwargs` are not past through is `MPLBACKEND=PS`, let's see if `SVG` works on travis (it works locally)

File: contact_map/contact_count.py [hunk 1]
```diff
@@ -98,19 +100,51 @@ def df(self):
...
+            msg = ("The number of pixels in the figure is insufficient to show"
+                   " all the contacts.\n Please save this as a vector image "
+                   "(such as a PDF) to view the correct result.\n Another "
+                   "option is to increase the 'dpi' (currently: "+str(dpi)+"),"
+                   " or the 'figsize' (curently: "+str((figwidth, figheight)) +
+                   ").\n Adviced minimum amount of pixels = "
+                   + str((self.n_x, self.n_y))+" (width, height).")
+            warnings.warn(msg, RuntimeWarning)
+
+    def plot(self, cmap='seismic', vrange=(-1.0, 1.0), with_colorbar=True,
```
Comment (dwhswenson) [inline contact_map/contact_count.py]: Go ahead and keep `vmin`/`vmax`. We'll tell codeclimate that this is an exception (in general, when most arguments are minor optional tweaks of behavior, I don't worry about the too-many-arguments complaint -- so in my mind, `with_colorbar` and `kwargs` don't really count). The reason to keep them is that `vmin`/`vmax` are the standard names in matplotlib, so we want to keep closer to standard matplotlib terminology.

File: contact_map/contact_count.py [hunk 2]
```diff
@@ -98,19 +100,51 @@ def df(self):
...
-        vmin : float
-            minimum value for color map interpolation; default -1.0
-        vmax : float
-            maximum value for color map interpolation; default 1.0
-
+        vrange : tuple
+            (minimum, maximum) value for color map interpolation;
+            default (-1.0, 1.0)
+        **kwargs
+            All aditional keyword arguments to be passed to the
```
Comment (dwhswenson) [inline contact_map/contact_count.py]: typo: `additional` (missing d)

File: contact_map/contact_count.py [hunk 3]
```diff
@@ -98,19 +100,51 @@ def df(self):
...
+        ypixels = dpi*figheight
+
+        # Check if every value has a pixel
+        if xpixels/self.n_x < 1 or ypixels/self.n_y < 1:
+            msg = ("The number of pixels in the figure is insufficient to show"
+                   " all the contacts.\n Please save this as a vector image "
+                   "(such as a PDF) to view the correct result.\n Another "
+                   "option is to increase the 'dpi' (currently: "+str(dpi)+"),"
+                   " or the 'figsize' (curently: "+str((figwidth, figheight)) +
+                   ").\n Adviced minimum amount of pixels = "
```
Comment (dwhswenson) [inline contact_map/contact_count.py]: "Recommended minimum" (better word choice, and typo in current spelling anyway)

File: contact_map/contact_count.py [hunk 4]
```diff
@@ -98,19 +100,51 @@ def df(self):
...
+        xpixels = dpi*figwidth
+        ypixels = dpi*figheight
+
+        # Check if every value has a pixel
+        if xpixels/self.n_x < 1 or ypixels/self.n_y < 1:
+            msg = ("The number of pixels in the figure is insufficient to show"
+                   " all the contacts.\n Please save this as a vector image "
+                   "(such as a PDF) to view the correct result.\n Another "
+                   "option is to increase the 'dpi' (currently: "+str(dpi)+"),"
+                   " or the 'figsize' (curently: "+str((figwidth, figheight)) +
```
Comment (dwhswenson) [inline contact_map/contact_count.py]: typo: should be "currently"

File: contact_map/contact_count.py (hunk 3 above)
Comment (sroet) [reply contact_map/contact_count.py]: like this, or without the `amount of pixels`?

**Comment by sroet** [comment]
> Sorry for long delay in reviewing; I was running the ESDW this week.

No problem, ran through your comments, please have another look.

File: contact_map/contact_count.py (hunk 3 above)
Comment (dwhswenson) [reply contact_map/contact_count.py]: with `amount of pixels`; I just meant that "Adviced" is wrong

**Comment by dwhswenson** [comment]
All looks good! Will merge once AppVeyor officially approves. (And for some reason codeclimate didn't even complain?)

**Comment by sroet** [comment]
> All looks good! Will merge once AppVeyor officially approves. (And for some reason codeclimate didn't even complain?)

I might have misinterpreted a `code complexity` complaint as `to many attributes`

**Comment by dwhswenson** [comment]
@sroet : In the end, this was not an API break, right? I think you undid the original API-breaking behavior. (Prepping a release right now.)

**Comment by sroet** [comment]
> @sroet : In the end, this was not an API break, right? I think you undid the original API-breaking behavior. (Prepping a release right now.)

Yes, this was only added functionality (pass through of `**kwargs` in the `plot` function to `matplotlib.subplots`) and bugfixes (prevents incorrect representations).

**Pull request merged**